java/arcs/core/host/ArcState.kt

```diff
@@ -190,6 +190,13 @@ data class ParticleState private constructor(val state: State) {
         val Failed_NeverStarted = ParticleState(State.Failed_NeverStarted)
         val MaxFailed = ParticleState(State.MaxFailed)
 
+        /** Creates [ParticleState] from serialized [toString] representation. */
+        fun fromString(serializedState: String) = serializedState.split('|', limit = 2).let {
```

Comment:
could this just be `serializedState.substringBefore('|')`?

Reply:
actually it's a bug, the cause is in it[1]